hazelcast/src/main/java/com/hazelcast/config/NearCacheConfigReadOnly.java

```diff
@@ -81,9 +81,4 @@ public NearCacheConfig setLocalUpdatePolicy(LocalUpdatePolicy localUpdatePolicy)
     public NearCacheConfig setEvictionConfig(EvictionConfig evictionConfig) {
         throw new UnsupportedOperationException("This config is read-only");
     }
-
-    @Override
-    public EvictionConfig getEvictionConfig() {
-        return super.getEvictionConfig().getAsReadOnly();
```

Comment:
I just added that getter two days ago, so this is not a huge drawback. I still have to figure out a solution for this "the `EvictionConfig` needs different sanity checks depending on the context" issue.